### PR TITLE
Feature/bootstrap

### PR DESCRIPTION
**Pull request by JohnVidler**
Bumping the samples repository to use [codal-bootstrap](https://github.com/lancaster-university/codal-bootstrap) to build rather than bare build.py, so we can update the build tools independently from the project sources.

Build function for users should be the same, although the readme may need updating to match the new functions available.

### CODE REVIEW

**Comment by microbit-carlos** [comment]
This is pending https://github.com/lancaster-university/codal/pull/45

**Comment by carlosperate** [comment]
We should also considered if this is needed before adopting bootstrap:
- https://github.com/lancaster-university/codal/issues/49

File: CMakeLists.txt [hunk 1]
```diff
@@ -182,18 +183,18 @@ if("${CODAL_DEFINITIONS}" STRGREATER "")
...
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -include \"${EXTRA_INCLUDES_PATH}\"")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -include \"${EXTRA_INCLUDES_PATH}\"")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -include ${EXTRA_INCLUDES_PATH}")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -include ${EXTRA_INCLUDES_PATH}")
 endif()
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -I\"${PLATFORM_INCLUDES_PATH}\"")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I\"${PLATFORM_INCLUDES_PATH}\"")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -I${PLATFORM_INCLUDES_PATH}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I${PLATFORM_INCLUDES_PATH}")
```
Comment (microbit-carlos) [inline CMakeLists.txt:191]: This is undoing PR https://github.com/lancaster-university/microbit-v2-samples/pull/10

**Comment by JohnVidler** [comment]
Closing this PR to draft a new one with the updated repo(s)

**Pull request closed without merging**